aptos-move/aptos-aggregator/src/resolver.rs

```diff
@@ -18,29 +28,89 @@ pub enum AggregatorReadMode {
 /// Returns a value of an aggregator from cache or global storage.
 ///   - Ok(..)       if aggregator value exists
 ///   - Err(..)      otherwise.
-pub trait AggregatorResolver {
+pub trait TAggregatorView {
+    type Identifier;
+
+    fn get_aggregator_v1_state_value(
+        &self,
+        id: &Self::Identifier,
+    ) -> anyhow::Result<Option<StateValue>>;
+
+    fn get_aggregator_v1_value(&self, id: &Self::Identifier) -> anyhow::Result<u128> {
+        let maybe_state_value = self.get_aggregator_v1_state_value(id)?;
+        // TODO: consider reviving Option<u128>?
+        let u128_bytes = maybe_state_value
+            .expect("Aggregator V1 cannot be deleted")
+            .into_bytes();
+        bcs::from_bytes(&u128_bytes)
+            .map_err(|_| anyhow::Error::msg("Failed to deserialize aggregator value to u128"))
+    }
+
+    // TODO: Nested options are used due to cyclic dependency. Aggregator crate needs refactoring.
```

Comment:
bad comment: we would have a cyclic dependency between aptos-aggregator and aptos-vm-types because the latter one has this…
```
type StateValueMetadataKind = Option<StateValueMetadata>;
```
the first option if value exists, the second if metadata exists

Reply:
Actually moved this to where state value is defined and got rid of that todo.